cpp/cmake/cpplib.py

```diff
@@ -391,55 +391,31 @@ def merge_clang_tidy_checks(orig_checks, new_checks):
     return ",".join(orig_checks)
 
 
-def do_merge_yaml(*files, **kwargs):
-    """Merge YAML files. The last argument is the destination file
+def do_merge_yaml(*files, output=None, transformers=None):
+    """Merge YAML files.
     """
     import yaml
 
-    transformers = kwargs.get("transformers", {})
-    out = files[-1]
-    ins = files[:-1]
-
-    outdated = not osp.exists(out) or osp.getmtime(out) < max(
-        (osp.getmtime(f) for f in ins)
-    )
-    if outdated:
-        data = {}
-        for file in ins:
-            with open(file) as istr:
-                for key, value in yaml.safe_load(istr).items():
-                    transform_func = transformers.get(key)
-                    if transform_func:
-                        data[key] = transform_func(data.get(key), value)
-                    else:
-                        data[key] = value
-        logging.info("writing file %s", out)
-        with open(out, 'w') as ostr:
-            yaml.dump(data, ostr, default_flow_style=False)
-    else:
-        logging.info("file %s is up to date, nothing to do.", out)
-    return True
+    data = {}
+    for file in files:
+        for key, value in yaml.safe_load(file).items():
+            transform_func = transformers.get(key)
+            if transform_func:
+                data[key] = transform_func(data.get(key), value)
+            else:
+                data[key] = value
+    logging.info("writing file %s", output)
+    yaml.dump(data, output, default_flow_style=False)
 
 
 def main(args=None):
     parser = argparse.ArgumentParser(description="Utility program")
-    subparsers = parser.add_subparsers(help='sub-command help')
-    merge_yaml = subparsers.add_parser('merge-yaml', help='Merge yaml files')
-    merge_yaml.add_argument("files", nargs='+', help="input files then output file")
-    merge_yaml.set_defaults(func=do_merge_yaml)
-
-    merge_yaml = subparsers.add_parser(
-        'merge-clang-tidy-config', help='Merge ClangTidy configuration files'
-    )
-    merge_yaml.add_argument("files", nargs='+', help="input files then output file")
-    merge_yaml.set_defaults(
-        func=functools.partial(
-            do_merge_yaml, transformers=dict(Checks=merge_clang_tidy_checks)
-        )
-    )
+    parser.add_argument("--clang-tidy", action="store_true", help="flag used if you want to merge clang tidy files")
```

Comment:
it doesn't look very consistent - as mentioned above. I preferred the previous implementation (although it could be cleaned up a bit too).

Reply:
In fact this is exactly the same command except we use a checker on a specific line of .clang _tidy.